### PR TITLE
Refactor - Use interface for rekor

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Use rekor interface instead of concrete implementation so that it can
  be mocked and tested.

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Comment by naveensrinivasan** [comment]
@laurentsimon 👀

File: pkg/mockentries.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import (
+	reflect "reflect"
+
+	runtime "github.com/go-openapi/runtime"
+	gomock "github.com/golang/mock/gomock"
+	entries "github.com/sigstore/rekor/pkg/generated/client/entries"
+)
+
+// MockEntriesClientService is a mock of ClientService interface.
+type MockEntriesClientService struct {
```
Comment (ianlewis) [inline pkg/mockentries.go:12]: Maybe provide a Makefile target or instructions somewhere on how to regenerate these?

File: pkg/mockentries.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
+package pkg
```
Comment (ianlewis) [inline pkg/mockentries.go:1]: Put generated mocks in a separate package.

**Comment by ianlewis** [comment]
I'm going to close this as it's pretty old. Feel free to resubmit the PR if we need this.

**Pull request closed without merging**